Nanotrasen Ranger-Class Aid Vessel (Power Rangers)

Latentish opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
![2022 07 11-01 26 51](https://user-images.githubusercontent.com/55075747/178222112-10578493-de45-46a5-a4d0-71c611cfb8b1.png)
![rangerclass](https://user-images.githubusercontent.com/55075747/178222114-c4b43e1e-a72f-4866-afdc-3d59e75ee8c7.png)

WELCOME TO THE POWER RANGERS. This fantastic ship has a crack team of work specialists led by the Lieutenant and the Commissioner.
The Commissioner is responsible for assigning jobs to the Loss Prevention team, dealing with wideband and holocalls, flying the ship, and organizing the duties of the non Loss Prevention staff.
Save ships as the power rangers! Upgrade your ship! Get allies by offering comms and telepads! Go on ship specific missions! Fight hostile syndicate! Protect your precious secret documents! Fill your vault with arms and loot! Do outpost missions!

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
This ship fills a relatively untouched niche amongst ships (amoung). It centers around a specialist team working together with the support of the scientist, miner, and commissioner. It's a cool nanotrasen vehicle that will hopefully get more people to play faction ships.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:phoaly
add: Adds the Nanotrasen Ranger-Class Aid Vessel
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Latentish committed:
Adds a bunch of shit

-Fuck you
-They're the power rangers
-Science ship???
-Ignore the honksquad suit it isn't real

ilikesaying commented:
![image](https://user-images.githubusercontent.com/65475697/175793695-9978e164-a3b6-4bad-a617-adc0d7dc6914.png)

Apogee-dev commented:
You may want to consider locking the power ranger gear behind access-locked doors windoors and./or lockers so the other crew members (commissioner) don't scoop their gear up. If you're using subtypes of the ERT jobs they'll have centcomm access and the captain won't, so use their job-specific access (ACCESS_CENT_STORAGE for engi, ACCESS_CENT_MED for med, ACCESS_CENT_CAPTAIN for commander and... well, ert sec doesn't get any unique access the other power rangers don't, unfortunately, but ACCESS_CENT_GENERAL will at least keep people who aren't power rangers out) to lock doors/windoors/lockers to a specific member of the team.

Latentish commented:
>You may want to consider locking the power ranger gear behind access-locked doors windoors and./or lockers so the other crew members (commissioner) don't scoop their gear up. If you're using subtypes of the ERT jobs they'll have centcomm access and the captain won't, so use their job-specific access (ACCESS_CENT_STORAGE for engi, ACCESS_CENT_MED for med, ACCESS_CENT_CAPTAIN for commander and... well, ert sec doesn't get any unique access the other power rangers don't, unfortunately, but ACCESS_CENT_GENERAL will at least keep people who aren't power rangers out) to lock doors/windoors/lockers to a specific member of the team.

I will most likely be using custom roles with custom accesses instead of normal ert access. Because they are supposed to be more minor team.

Latentish committed:
more

adds more shit

Apogee-dev commented:
>I will most likely be using custom roles with custom accesses instead of normal ert access. Because they are supposed to be more minor team.

Well, to be fair, giving them centcom access isn't a huge deal since it isn't really used anywhere, and their all-access similarly isn't the end of the world because ship-specific access exists to keep them from pillaging other ships' armories or whatever. for the most part the original ERT access would just be a convenient way of keeping the commissioner out of their gear since (assuming that's the captain role) commissioner would automatically get all-access.

also whoops i fucked up and typed in the wrong field sorry sylv.

Latentish committed:
Adds a lot of shit.

- Adds subtypes of 5 pairs of outfits
- Adds new subtypes of the ERT hardsuits that aren't excessively strong
- Adds a couple mindshield locked guns
- Hides the clown hardsuit
- Adds most of the equipment for the Power Rangers.

Latentish commented:
> > I will most likely be using custom roles with custom accesses instead of normal ert access. Because they are supposed to be more minor team.
> 
> Well, to be fair, giving them centcom access isn't a huge deal since it isn't really used anywhere, and their all-access similarly isn't the end of the world because ship-specific access exists to keep them from pillaging other ships' armories or whatever. for the most part the original ERT access would just be a convenient way of keeping the commissioner out of their gear since (assuming that's the captain role) commissioner would automatically get all-access.
> 
> also whoops i fucked up and typed in the wrong field sorry sylv.

I was thinking of the commissioner being sort of like the HoP on HRP servers where they do more than just change access. While the commander would be the actual owner. But it isn't a big deal either way, to make new accesses.

Latentish committed:
Adds wires and atmos

Adds areas, atmos, areas.

Latentish committed:
subtypes and accesses and FUCK

-Adds IDS job loadouts, and most of the remaining bits and pieces.
-Adds some fucking issue with the ship I haven't divined yet.

Latentish commented:
Where the sweet mcfuck is the error

ZephyrTFA commented on _maps/configs/nanotrasent_powerrangers.json:
```diff
@@ -0,0 +1,47 @@
+			"outfit": "/datum/outfit/job/lp/commissioner",
+			"slots": 1
+		},
+		"AI": 1,
```
this isnt a valid slot entry

Latentish replied:
Thank you.

ZephyrTFA commented on shiptest.dme:
```diff
@@ -2165,6 +2165,7 @@
 #include "code\modules\jobs\job_types\head_of_security.dm"
 #include "code\modules\jobs\job_types\janitor.dm"
 #include "code\modules\jobs\job_types\lawyer.dm"
+#include "code/modules/jobs/job_types/lp_jobs.dm"
```
did you manually put this in the DME?

Latentish replied:
I fixed that, and yes.

Latentish committed:
Found the issue also

-Adds a new pda type for the comm
- Properly names the map and removes the duplicate

Latentish committed:
I hate access and job code

-Why does it scream
-IT doesn't work
-I'm in misery
- Why doesn't it getaccess()

Latentish committed:
Removes the ai slot

Goodbye

Latentish commented:
Shitcode

Latentish commented:
![image](https://user-images.githubusercontent.com/55075747/176631556-1cc1aa60-6b66-43e0-b35b-b6ff796b270f.png)

Could I please get a larger brain than I have to look at my code? Jobcode is arcane and it seems that I have nothing to go off of other than jobs explicitly meant for stations, with accesses I don't intent to use. I can't seem to find any other jobs made without a subtype so I'm rather confused on how this all works out properly. There's no "department" or anything to relate it to and VSC keeps saying that it can't run null.get_access()
When I run a test of the game it manages to spawn me in with a proper id with proper access. But it takes forever and everything lags.

Latentish commented:
I also want to  point out that this check is bullshit because I checked all these and none of the pipes around them are disconnected. None of them are actually. Everything is connected properly from what I can tell.
![image](https://user-images.githubusercontent.com/55075747/176634621-498e4aa1-378d-47dd-be39-5ea47b57c68b.png)

Fikou commented on code/game/objects/items/cards_ids.dm:
```diff
@@ -934,3 +934,68 @@ update_label()
 	icon_state = "sec_budget"
+
+//Power rangers IDs
+//lieutenant locked - 2398
```
these numbers must be defines

Fikou commented on code/modules/clothing/spacesuits/miscellaneous.dm:
```diff
@@ -469,3 +469,83 @@ Contains:
+	name = "Loss Prevention Security Hardsuit"
+	desc = "The best of the best security staff get assigned to the ERT. Second best are given this Hardsuit as a part of the LP Team."
+	icon_state = "ert_security"
+	item_state = "ert_security"
```
you don't need to override these, the parent already set them, same case for the rest of those

Latentish replied:
The outfits are subtypes of the commander lp. They all share the commander suit. So the rest do actually need to be defined. But the first one probably foesn't.

Fikou commented on code/modules/clothing/spacesuits/miscellaneous.dm:
```diff
@@ -469,3 +469,83 @@ Contains:
+
+//The funny power rangers.
+/obj/item/clothing/suit/space/hardsuit/ert/lp
+	name = "Loss Prevention Lieutenant Hardsuit"
```
item names are usually lowercase, is there any specific reason for these not to be?

Latentish replied:
Funny titles. I suppose its just habitual. No not really in particular

Latentish replied:
These are mostly titles so capitalization isn't out of place I would say.

Fikou commented on code/modules/clothing/spacesuits/miscellaneous.dm:
```diff
@@ -469,3 +469,83 @@ Contains:
+	hardsuit_type = "ert_medical"
+
+/obj/item/clothing/head/helmet/space/hardsuit/ert/lp/clown
+	name = "LP Funny Hardsuit Helmet"
```
this one (and the suit) says LP instead of Loss Prevention, is that intended?

Latentish replied:
Uhhhhh I will double check..

Latentish replied:
Yes this is intentional.

Latentish replied:
If it's LP and not Loss prevention, it could stand for anything else with those letters. Laughter Protocall? I'll likely write something into the description for it later.

Fikou commented on code/game/objects/items/cards_ids.dm:
```diff
@@ -934,3 +934,68 @@ update_label()
+//vault = 5592
+//bridge = 7921
+//AI = 5832
+/obj/item/card/id/lpengie
```
why do these id's need to exist? won't the job datum apply the required access?

Latentish replied:
I don't know. It's been spitting (cannot do getaccess()) each time. Theres no real eqxamples for what I'm trying to do. I could very seriously use some help with the jobcode.

Latentish replied:
I removed the individual accesses from the ID's and made defines for the numbers, but I've kept the ID's with the unique titles. Although that might also get handled by the code? I'm still a bit confused how to handle the jobcode and outfit code.

Latentish replied:
This should have been fixed.

Fikou commented on code/modules/jobs/job_types/lp_jobs.dm:
```diff
@@ -0,0 +1,152 @@
+/datum/job/lp/lieutenant
```
1. won't you want to define the job's outfits to be the ones you made?
2. shouldn't these be subtypes of their counterparts (like engineering specialist being a subtype of station engineer?)

Latentish replied:
I'm not sure I understand your first one. Yes I'd like my outfits to be applied, I've been having difficulty with that.
Ehhhhhhhh they're all funny subtypes of their commander. I've already put in the work to make them so, I' don't realqly see a reason to make them subtypes of the normal hardsuits. They don't share much with them (including stats).

Latentish replied:
1. Understanding what you mean now, I've actually attached the outfit to the job.
2. Not for the LP, I wanted to make wholly unique jobs and outfits for them so I made them subtypes of /lp/.

Latentish replied:
This should be fixed now.

Latentish committed:
Adds defines

- Adds actual defines instead of arbitrary numbers.

Latentish committed:
Fixes the suit

-Replaces the deprecated suit with the proper version

Latentish committed:
Attaches the outfits

- Adds the datums for the outfits to their respective jobs.

Latentish commented:
Yeah so I'm having trouble figuring out this error I get.
![image](https://user-images.githubusercontent.com/55075747/176854215-f4e00264-cafc-4217-9c8e-fafc7eeb3e72.png)

Latentish committed:
Fixes pipes and wires

-The shiptest can breathe again
-The shiptest has electricity

Latentish committed:
FIXES THE ERROR

-Adds proper job types
-Adds factions to the lp master
-Fixes the getaccess error

Latentish committed:
Rearranges pipes

-Hopefully solves integration test issue
-Moves pipes under an ORM to not be there.

Latentish commented:
![image](https://user-images.githubusercontent.com/55075747/176948005-30bebb75-a7e1-469f-8a86-3f089064eafe.png)
What are you fucking on G? where WHERE

Latentish commented:
FOUND IT FUCK

Latentish committed:
More or less finishes the ship

- Adds the subtypes for doors and windows
-adds closet subtypes
-adds more access to the jobs
-Adds more decals

Latentish committed:
Adds missions

-Adds mission holodisks and a spawner for them.
- Tweaks door accesses
- Fixes closets not having doors.

Latentish committed:
Tweaks lightswitches

-Light switches are in more sane locations.

Vexylius reviewed (approved):
y es, looks good

Latentish committed:
Fixed

-Fixes power
-Fixes Shutters

Latentish committed:
Buttons and solars

-Fixes a button
-Further unwires solars

Latentish committed:
Adds blast doors tot he thrusters

titular

Latentish committed:
Direction port change

-Changes the default direction of the ship to be correct
-Redoes the NT logo to be white instead of whatever it was before

Latentish committed:
Fixes engines

Adds the missing wires to the ion thrusters.

Latentish committed:
Fixes ammo in comm's closet

titular

Apogee-dev reviewed (approved):
Looks good to me, any outstanding issues I've raised in the past have been resolved and I don't see anything obviously wrong in its current state. Should be ready for testmerge.

Latentish committed:
LP Patch

-Fixes accesses
-Redoes some room
-Adds missing processor unit and aicore module

Latentish committed:
Why did this disappear

-This pump disappeared.

Latentish committed:
Forgot to add the sec job

-Readds the sec job

Latentish committed:
small fixes

-Replaces the combat medkit with an advanced one.
- Moves some glass.
- Fixes rechargers
-gives the medic lp a crew pinpointer

Latentish committed:
fucky windows

-Replaced the normal windoors with the correct access locked ones.

Zevotech commented:
- Combat medkit in the medical LP locker has a syndicate defibrillator, night vision medical hud and a combat stimulant injector in it. Replacing the medkit itself with an elite nanotrasen belt defib and a different combat stimulant injector or medical hud implanter (like the one the CMO has) would make this slightly more accurate.
- The normal white medkit in the medical LP locker is the wrong subtype and starts out completely empty
- The airlocks do not function/barely function and may need more vents
- There is only one set of surgical tools
- The fridge has an overabundance of grains and grain based foods, and the bar area has no microwave to cook anything with.
- There is no Security LP Officer slot that players can join as.

Latentish committed:
Ranger Patch

- Changes airlock vents with large volume vents
- Redoes the lockers of the Power Rangers
- Adds a fire suit that has no defense, but is immune to heat
- Adds a special medkit for the Medical Ranger
- Changes some of the food
- Adds a second set of surgical tools for science
- Gives the Commissioner vault access

Latentish committed:
Food?

-Adds a knife
-Makes the menu more vox and sarathi friendly.

Latentish committed:
Nodes...

Adds proper nodes to the engines.

Latentish committed:
Handful

- Adds small slowdown to the LP hardsuits
- Moves some things in the kitchen around
- Changes the kitchen locker
- Adds an empty-AI shell to the AI sat
- Removes ai and cyborg uploads
- Changes quantum pad holotape to require a teleporter instead
-Adds a jukebox

Latentish committed:
quick fix

-Fixes a wrong typepath

Latentish committed:
Reduces files affected

- Adds folders to the code/data/ folder for ruin and shuttle data
- Puts most subtypes and data for the ship in one file
- Removes the radio and ai shell (they don't work)

Latentish committed:
Missed a few things

-Correctly adds and removes code from extraneous files

Latentish committed:
decal work

-Makes the foyer prettier

Latentish committed:
Fixes the Comm's access

-Titular

Fikou commented on code/datums/lp_mission_holotapes.dm:
```diff
@@ -0,0 +1,299 @@
+	desc = "A holodisk containing a combat mission for the LP."
+	preset_image_type = /datum/preset_holoimage/commissioner
+	preset_record_text = {"
+	NAME Commissioner Gorre Donn
+	SAY This mission, should you choose to accept it, revolves around a combat objective. The LP will have to get their hands dirty for this one.
+	DELAY 25
+	SAY The LP is to respond to a hostile threat from a Syndicate vessel. In such action they are to kill or otherwise subdue the primary fighting force.
+	DELAY 25
```
tab these one more please

Fikou replied:
for all of em

Latentish replied:
This is how they're formatted by mark and anturk in the holocall file.
![image](https://user-images.githubusercontent.com/55075747/178066071-3c1590e0-2401-4a38-9ff6-dd71a4b954f0.png)
![image](https://user-images.githubusercontent.com/55075747/178066170-fccdf97d-61e6-43f9-8ace-a05038eb4d0d.png)

Fikou replied:
sure i guess

Fikou commented on code/modules/jobs/job_types/lp_jobs.dm:
```diff
@@ -0,0 +1,195 @@
+	courierbag = /obj/item/storage/backpack/messenger/com
+
+
+/datum/job/scientist/lp
```
why do scientist and miner have lp subtypes and all the other guys are subtypes of lp

Fikou replied:
nvm i think i know what youre getting at

Latentish replied:
Yeah, the comm and the power rangers are all subtypes of the Lieutenant to give them special access.
The rest of the standing crew are subtypes of their normal jobs to compliment the ship.

Fikou commented on code/modules/jobs/job_types/lp_jobs.dm:
```diff
@@ -0,0 +1,195 @@
+
+/datum/outfit/job/miner/lp
+	ears = /obj/item/radio/headset/binary
+	id = /obj/item/card/id/minelp
```
what is the purpose of these ids? why not just use the standard ones?

Latentish replied:
Description, icon, name.
![image](https://user-images.githubusercontent.com/55075747/178065861-90f94355-b49b-40e3-ab99-64ba79944797.png)

Fikou replied:
The name will get set by the job.
The desc is useless.
The icon you can just use centcom id cards

Latentish replied:
I'd still like to have the nanotrasen icon, but I can just find which one that is and change their ID to that, since they'll all have their access written properly.

Man... Fuck jobcode.

Latentish replied:
I looked in the code and the mapper for any ID that uses the "nanotrasen" icon, they all use the centcomm officer one.
The one on the right has stripes that denote authority and a description on the card that denotes being sent from centcomm. I don't want that. The one on the right is only used in these ID's.
![image](https://user-images.githubusercontent.com/55075747/178189845-f84c8ff0-1a4e-4f1a-9c26-8e37c52edebd.png)

Latentish replied:
Realizing that the icons won't apply properly, I just removed these ID's and am using their bases.

Latentish committed:
LP Patches

- Adds an air recycling system.
- Removes redundant pumps.
- Replaces the security protolathe with a research protolathe.
- Adds accesses to the access console.
- fixes weird gloves in outfits.
- Removes shitcode ID cards.
- Adds an access console to the commissioner's office.

Latentish committed:
DOES THIS MAKE YOU HAPPY?

HUH RUN LINTERS DOES THIS MAKE YOU SMILE?

Latentish committed:
Readds the note

-tries desperately to appease the run linters

Latentish committed:
Switches the note

- Makes the note a holodisk instead.
- Puts a holopad in the AI core.

Latentish commented:
Let's _hope_ that giving the budget _ert_ ship a science/production _protolathe_, won't be a **nightmare**.

Apogee-dev commented:
There's carpet underneath the airlocks and this cannot stand

Latentish committed:
Minor update

-Changes the carpet tile under the one airlock to a plating.
- Adds 3 sets of air and fire emergency wallclosets.

RuneQuester commented:
Commissioner doesn't appear to have their own hardsuit(?) though this could be intentional and Im sorry for typing this if it is.
No sabre sheath for the commissioner (personal request)
No like...signified, captain armor for the commissioner either. Just an armor vest??

Latentish commented:
> Commissioner doesn't appear to have their own hardsuit(?) though this could be intentional and Im sorry for typing this if it is. No sabre sheath for the commissioner (personal request) No like...signified, captain armor for the commissioner either. Just an armor vest??

All of that is intentional. The commissioner isn't a captain. They get their own office because it helps for them to do their job better, that job basically being the team's handler and HR person. They aren't the final say on anything, that's the Lieutenant, though in most rounds I've played, the Lieutenant and Commissioner have worked together pretty flawlessly.
The commissioner's doesn't go into the field unless they have to, in which case, there is a secret 5th hardsuit on the ship somewhere as a secret. They also have access to the mining hardsuits. But they don't get access to any of the LP gear that isn't in the vault.

RuneQuester commented:
Also, the first person 

> > Commissioner doesn't appear to have their own hardsuit(?) though this could be intentional and Im sorry for typing this if it is. No sabre sheath for the commissioner (personal request) No like...signified, captain armor for the commissioner either. Just an armor vest??
> 
> All of that is intentional. The commissioner isn't a captain. They get their own office because it helps for them to do their job better, that job basically being the team's handler and HR person. They aren't the final say on anything, that's the Lieutenant, though in most rounds I've played, the Lieutenant and Commissioner have worked together pretty flawlessly. The commissioner's doesn't go into the field unless they have to, in which case, there is a secret 5th hardsuit on the ship somewhere as a secret. They also have access to the mining hardsuits. But they don't get access to any of the LP gear that isn't in the vault.

I REALIZED THATS WHY I DELETED IT AAAAAAAAA IM SORRY

RuneQuester commented:
However a legitimate question/complaint. Tcomms doesnt appear to...work, roundstart? everything says NULL.

Latentish commented:
> However a legitimate question/complaint. Tcomms doesnt appear to...work, roundstart? everything says NULL.

You have to set up TComms at roundstart. This is done by linking all the machines together with a multitool, setting the radio channels you want, naming the communications server to what you want it to be, and then searching for it on the console to monitor it if you wanted.

Fikou commented on code/modules/jobs/access.dm:
```diff
@@ -372,6 +372,26 @@
+		if(ACCESS_LP_MEDIC)
+			return "LP Medical Specialist"
+		if(ACCESS_LP_ENGINEER)
+			return "Lp Engineering Specialist"
```
```suggestion
			return "LP Engineering Specialist"
```

Fikou commented on code/game/MapData/shuttles/nanotrasen_powerrangers.dm:
```diff
@@ -0,0 +1,373 @@
+/obj/item/clothing/suit/space/hardsuit/ancient/lp
+	name = "RIG heat suit"
+	desc = "A fully heat resistance suit based on an early RIG hardsuit prototype. It sacrifices armor of any kind for intricate heatsinks. It remains rather bulky as a result."
+	armor = list("melee" = 1, "bullet" = 1, "laser" = 1, "energy" = 1, "bomb" = 50, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 75)
```
why 1

Latentish replied:
Because its a maxtemp immune suit, it serves no other purlose than to withstand that. It has almost no defense in exchange for that property.

Fikou replied:
Then do 0

Latentish committed:
Update code/modules/jobs/access.dm

Yeah

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

Latentish committed:
Tweaks heatsuit

- Removes the armor from the heatsuit.
